### DISCUSSION TITLE
k-betweenness

### DISCUSSION THREAD

**Discussion by sdelandtsheer** [networkx/networkx#5942] [Algorithms]
I've had the idea of bounding the betweenness calculations (to increase relevance and decrease computational time). I just found out k-betweenness actually exists here: https://www.centiserver.org/centrality/K-Betweenness_Centrality/
Am I right that it is not implemented as such in the current release?
Anyone has information on a working Python implementation?

Thanks

**Comment by dschult** [comment]
The NetworkX `betweenness_centrality` function has a second input parameter named `k` to restrict paths to length `k`. So, I think it is implemented here in NetworkX

**Comment by sdelandtsheer** [comment]
@dschult In the current function the parameter `k` is used to sample nodes, not to restrict the length of paths.
At least this is how I understand [this documentation](https://networkx.org/documentation/networkx-1.10/reference/generated/networkx.algorithms.centrality.betweenness_centrality.html).

**Comment by dschult** [comment]
Yes -- You are correct and I read it too fast (reading too fast when you want to see something makes me tend to see it even when its not always there).  Sorry about that.

In the case where we use unweighted edges, this should be easy to implement. The BFS routine used to compute shortest paths could include a cutoff for the path length searched.  But for weighted paths, you need to look at the entire graph before knowing whether the path is more than or less than a cutoff. So I don't think a general weighted method would help speed this up much. For the unweighted case, look at the source code for `betweenness_centrality` and either replace `_single_source_shortest_path_length` with `nx.single_source_shortest_path_length`or scroll down a little and change `_single_source_shortest_path_length` to stop searching when a cutoff length is reached.

Meanwhile, someone else might know of another implementation.  Anyone?

**Reply by sdelandtsheer** [reply]
Thanks, that helps a lot.
In the meantime I looked at the formula again ([here](https://www.centiserver.org/centrality/K-Betweenness_Centrality/)) and I also understood it wrong ;-) The metric that is described is for 'short paths', that is paths with a length smaller than k + the length of the shortest path. So yet another way to compute betweenness.
In my case I only deal with unweighted undirected networks.
So, still not sure my idea of restricting the pairs of nodes to a max path length (kind of localized betweenness) exists anywhere, which makes me wonder if it is a good idea at all.
